vfs/directory.go

```diff
@@ -257,6 +335,45 @@ func bulkUpdateDocsPath(c *Context, olddoc *DirDoc, newpath string) error {
 	return err
 }
 
+func fetchChildren(c *Context, parent *DirDoc) (files []*FileDoc, dirs []*DirDoc, err error) {
+	var docs []json.RawMessage
+	sel := mango.Equal("folder_id", parent.ID())
+	req := &couchdb.FindRequest{Selector: sel, Limit: 10}
+	err = couchdb.FindDocs(c.db, FsDocType, req, &docs)
+	if err != nil {
+		return
+	}
+
+	var t struct {
+		Type string `json:"type"`
+	}
+
+	// @NOTE: this unmarshaling technique is very slow...
```

Comment:
The only way I found to do that was to have a wrapper struct `{ Type string, Data json.RawMessage }`, and I wanted to avoid nested struct for now...

Also the slowness is relative.. I measured ~18ms for 1000+ files, which is still quite low compared to CouchDB response which took about 300ms.